test/end2end_test.go

```diff
@@ -2281,6 +2289,56 @@ func testClientRequestBodyErrorCancelStreamingInput(t *testing.T, e env) {
 	})
 }
 
+func TestFailFastRPCRetryOnCredentialTimeout(t *testing.T) {
+	const minConnectTimeout = time.Millisecond
+	defer internal.TestingSetMinConnectTimeout(minConnectTimeout)()
+
+	before := grpc.DefaultBackoffConfig.BaseDelay
+	grpc.DefaultBackoffConfig.BaseDelay = 100 * time.Millisecond
+	defer func() {
+		grpc.DefaultBackoffConfig.BaseDelay = before
+	}()
+
+	te := newTest(t, env{name: "slow-tls", network: "tcp", security: "tls"})
```

Comment:
One side effect is that `minConnectTimeout` is shared by all tests. Other tests may fail if they run in parallel with this test.